doc/manual/rule_uid_schema.md

```diff
@@ -0,0 +1,139 @@
+
+## Example
+
+Lets start with an example for a rule UID, and lets break down its concepts. The example references a rule listed in the subsection **Rules** of one of the ASAM Standards for OpenDRIVE. The Rules subsections were introduced for the first time in revision 1.6 of the standard. **Please notice that the full name of the rule may change in the future**.
+
+> [ASAM OpenDRIVE Standard 1.6.0, Chapter 7.2 - Road Reference Line](https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_road_reference_line), subsection Rule, first point: 
+> > **Each road shall have a reference line**
+
+we can identify the following concepts:
+
+ * **emanating entity**: ASAM `asam.net`
+ * **reference standard**: OpenDRIVE `xodr`
+ * **definition settings**: the rule firstly appeared in `1.6.0`, the first revision to introduce the subsections rules
+ * **Full name**:
+    * **rule-set**: Road Planar Geometry definitions `road.planview.geometry`
+    * **name**: summarizes the content of the rule, e.g. `ref_line_exists`. It is always the last group in dot-notation strings.
+ * **version**: can be set to `1` 
+
+The complete UID would be something like:
+
+```
+asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:1
+-------- ---- ----- ---------------------- --------------- -
+ |        |    |     |                      |              |
+ entitity | version  |                    name        rule-version
+       standard    rule-set
+```
+
+or without including the rule version:
+
+```
+asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists
+-------- ---- ----- ---------------------- ---------------
+ |        |    |     |                      |           
+ entitity | version  |                    name        
+       standard    rule-set
+```
+
+Regarding the validation for the schema, **only the the emanating concept and the name are required**. Third party entities are not required to declare reference standard, definition settings, rule-set and version. Still, separation characters cannot be omitted. ASAM rules shall always define all concepts; rule version is still optional.
+
+> Example: if `example.com` entity defines it's own rules, it may be something like: `example.com:::custom_rule` or `example.com:qc::custom_rule:2` and the UID should be considered valid. **However, entity should use ASAM standard and definition setting concepts in its custom rules.**
+
+## Query rules UID
+
+The proposed formalism allow to perform query on rules (or set of rules) using UNIX style wildcards notation (i.e., python `fnmatch` module). A query engine implementation shall return the most recent rule version available, when not stated explicitly in query string.
+
+Examples: 
+
+ * query all the OpenDRIVE geometrical rules that are not in version `>= 2` of the standard: `asam.net:xodr:1.?.0:*geometry.*` matches `asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:1`
```

Comment:
```suggestion
 * query all the OpenDRIVE geometrical rules that are not revised (version `< 2`): `asam.net:xodr:1.?.0:*geometry.*` matches `asam.net:xodr:1.6.0:road.planview.geometry.ref_line_exists:1`
```

Uh!

This whole story of the rule version is really dangerous, or difficult to understand. It looks like you stepped in our own trap. I understand that the standardization project needs this to make clear that only the prosa text was changed, not the meaning of the rule. From the QC point-of-view IMHO we just need to query or to work with the rules, not their version. I don't see a use case where we need to know the version of the rule. If you agree, then I would suggest to write that the optional version number is just for the documentation of the rules, but has no meaning for the framework.

Reply:
The example refers to the version **of the standard**, not the version of the rule (fnmatch for `1.?.0` matches `1.6.0` but not `2.0.0`). If the exampl is not clear I can rewrite it or eliminate it.

Reply:
I agree to enforce a version of the rule (not the standard) only for documentation purposes, and I agree that has no meaning from the framework.

Reply:
Example removed. I kept only the second example, which is clearer and on existing standards